src/app/carbon-estimator-form/carbon-estimator-form.component.spec.ts

```diff
@@ -50,14 +50,21 @@ describe('CarbonEstimatorFormComponent', () => {
   describe('headCount()', () => {
     it('should not return null once the component is initialized', () => {
       component.ngOnInit();
-      expect(component.headCount).not.toBe(null);
+      expect(component.headCount === null).toBeFalse();
```

Comment:
Any particular reason to change these to `=== null`? I would have thought the previous method would give a more understandable error - `Expected null not to be null.` vs  `Expected true to be false.`

Possible they're both not that helpful, could add
`expect(component.headCount).withContext('headCount').not.toBe(null);`
which outputs
`headCount: Expected null not to be null.`

Reply:
The issue was that it wouldn't compile using `.not.toBe(null)` because I've coerced the return values of the methods to not include `null` by adding a `!` after the `.get` because they should never return `null` once the form group has been created in `ngOnInit`. This was to avoid needing null checks when using the getters/to not have to pass a nullable argument to the pipe. Using `=== null` was a workaround for the tests. I'm slightly surprised it doesn't complain about the fact that the return type doesn't include `null` here as well. I'll see if I can improve the error message and add a comment if not.

Reply:
That's odd, I thought it might be around the coerced null but it didn't seem to have a problem with it for me.

Reply:
There may also be an option to suppress the compiler error on these lines and use `not.toBe(null)`.

Reply:
It seems fine with `.not.toBeNull()` so I've changed it to that. There is also [@ts-expect-error](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments) which would have suppressed the error for `.not.toBe(null)`.

Unfortunately, the pipe throws an error when a getter is broken which causes all the tests for the component to fail without a clear message. I'm not sure what to do about that.

Reply:
It's also causing the unit test suite to hang when they fail.

Reply:
Ah, well maybe you can't expect to be able to deal with everything - to reach that point I assume you had to deliberately return null while saying that it could never be null?

I guess the options are to allow the pipe to take null and immediately return false or try to mock out the pipe for the unit test but that seems a bit extreme.